fineract-provider/src/main/java/org/apache/fineract/portfolio/self/pockets/service/PocketAccountMappingReadPlatformServiceImpl.java

```diff
@@ -78,7 +80,7 @@ public PocketAccountMappingData retrieveAll() {
     public boolean validatePocketAndAccountMapping(Long pocketId, Long accountId, Integer accountType) {
         final String sql = "select count(id) from m_pocket_accounts_mapping mapping where pocket_id = ? and account_id = ? and account_type = ?";
         try {
-            return this.jdbcTemplate.queryForObject(sql, Boolean.class, pocketId, accountId, accountType);
+            return TRUE.equals(this.jdbcTemplate.queryForObject(sql, Boolean.class, pocketId, accountId, accountType));
```

Comment:
Not the best solution:
Since it is executing a select with count, there will be always a result. THe EmptyResultDataAccessException will never be thrown.
Also mapping the count result to boolean is questionable. If we are touching this, we should make it better!